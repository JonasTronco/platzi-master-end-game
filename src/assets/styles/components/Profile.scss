@import '../main.scss';

.profile {
  &__user-data {
    @include card;
    width: 100%;
    padding: $space_x6;
  }
}
