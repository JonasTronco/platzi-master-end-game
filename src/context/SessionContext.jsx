import { useState, useEffect, createContext, useMemo, useContext } from 'react';
import Router from 'next/router';

const SessionContext = createContext();
export function SessionProvider(props) {
  const [session, setSession] = useState(null);

  useEffect(() => {
    try {
      setSession(JSON.parse(localStorage.getItem('session')));
    } catch (error) {
      Router.push('/');
    }
  }, []);

  const value = useMemo(
    () => ({
      session,
    }),
    [session],
  );

  return <SessionContext.Provider value={value} {...props} />;
}

export function useSession() {
  const context = useContext(SessionContext);
  if (!context) {
    throw new Error('No Session Context Provider');
  }
  return context;
}
