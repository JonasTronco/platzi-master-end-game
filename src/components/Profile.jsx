import PropTypes from 'prop-types';
import useData from '../hooks/useData';
import { useSession } from '../context/SessionContext';
import '../assets/styles/components/Profile.scss';
import User from './User';
import UserData from './UserData';

function Profile({ userID }) {
  const { session } = useSession();
  const URL = `${process.env.NEXT_PUBLIC_API_URL}/users/${userID}`;
  const { data, loading, error } = useData(session.token, URL);

  if (loading) {
    return <div className="loader" />;
  }

  return data && data.data ? (
    <div className="profile">
      <User user={data.data} />
      <section className="profile__user-data">
        <UserData user={data.data} />
      </section>
    </div>
  ) : (
    ''
  );
}

Profile.propTypes = {
  userID: PropTypes.string.isRequired,
};

export default Profile;
