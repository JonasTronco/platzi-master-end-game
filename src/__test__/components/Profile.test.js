import Profile from '../../components/Profile';
import { create } from 'react-test-renderer';
import { SessionProvider } from '../../context/SessionContext';
describe('Profile Component', () => {
  test('Should render Profile component', () => {
    const component = create(
      <SessionProvider>
        <Profile userID='' />
      </SessionProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
