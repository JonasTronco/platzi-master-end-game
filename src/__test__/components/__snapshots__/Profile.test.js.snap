// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Component Should render Profile component 1`] = `""`;
